torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,215 @@
+  if (op_name == kConv2dOp) {
+    if (extra_args.find(kInputSize) == extra_args.end()
+        || extra_args.find(kWeightSize) == extra_args.end()) {
+      TORCH_WARN("Calculating flops for aten::conv2d requires input_size and weight_size in saved arguments.");
+      return 0;
+    }
+    auto input_sizes_ref = extra_args.at(kInputSize);
+    auto kernel_sizes_ref = extra_args.at(kWeightSize);
+    if (!input_sizes_ref.isIntList() || !kernel_sizes_ref.isIntList()) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires input and weight tensor sizes.");
+      return 0;
+    }
+
+    const std::vector<int64_t> input_sizes = input_sizes_ref.toIntVector();
+    const std::vector<int64_t> kernel_sizes = kernel_sizes_ref.toIntVector();
+    if (input_sizes.size() != 4 || kernel_sizes.size() != 4) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because both input and weight must be size 4.");
+      return 0;
+    }
+    // format of the input is defined in torch.nn.quantized.functional.conv2d()
+    uint64_t minibatch = 0, in_channels = 0, input_h = 0, input_w = 0;
+    uint64_t out_channels = 0, kernel_h = 0, kernel_w = 0;
+    const uint64_t conv2d_multiply_factor = 2;
+    std::tie(minibatch, in_channels, input_h, input_w) = std::make_tuple(input_sizes[0], input_sizes[1],
+                                                                         input_sizes[2], input_sizes[3]);
+    std::tie(out_channels, std::ignore, kernel_h, kernel_w) = std::make_tuple(kernel_sizes[0], kernel_sizes[1],
+                                                                              kernel_sizes[2], kernel_sizes[3]);
+
+    // grouping is NOT properly handled yet
+    return conv2d_multiply_factor * minibatch * input_h * input_w * kernel_h * kernel_w * in_channels * out_channels;
+  } else if (op_name == kGemmOp) {
+    if (extra_args.find(kMat1Size) == extra_args.end()
+        || extra_args.find(kMat2Size) == extra_args.end()) {
+      TORCH_WARN("Calculating flops for aten::mm requires mat1_size and mat2_size in saved arguments.");
+      return 0;
+    }
+    auto mat1_sizes_ref = extra_args.at(kMat1Size);
+    auto mat2_sizes_ref = extra_args.at(kMat2Size);
+    if (!mat1_sizes_ref.isIntList() || !mat2_sizes_ref.isIntList()) {
+      TORCH_WARN("Failed to compute flops for op aten::mm because it requires mat1_size and mat2_size to be IntList.");
+      return 0;
+    }
+
+    std::vector<int64_t> mat1_size = mat1_sizes_ref.toIntVector();
+    std::vector<int64_t> mat2_size = mat2_sizes_ref.toIntVector();
+    if (mat1_size.size() == 0) {
+      return 0;
+    } else {
+      int64_t overlap_dim = mat1_size.back();
+      uint64_t flops = 1;
+      for(int64_t dim : mat1_size) {
+        flops *= dim;
+      }
+      flops /= overlap_dim;
+      for(int64_t dim : mat2_size) {
+        flops *= dim;
+      }
+      return flops;
```

Comment:
Shouldn't we have an extra factor of 2 to account for multiplication and addition?